radix-engine/src/engine/kernel.rs

```diff
@@ -501,8 +452,28 @@ where
             }
         }
 
+        let mut call_frame = self.call_frames.pop().unwrap();
+
+        // Auto drop locks
+        for (_, lock) in call_frame.drain_locks() {
+            let SubstateLock {
+                pointer: (node_pointer, offset),
+                ..
+            } = lock;
+            if !(matches!(offset, SubstateOffset::KeyValueStore(..))
+                || matches!(
+                    offset,
+                    SubstateOffset::ResourceManager(ResourceManagerOffset::NonFungible(..))
+                ))
```

Comment:
This logic is duplicated a few times. Suggest we add a helper method  `is_key(&self) -> bool` to `SubstateOffset`